crates/orchestrator/src/lib.rs

```diff
@@ -637,10 +695,26 @@ pub use pjs_helper::PjsResult;
 
 #[cfg(test)]
 mod tests {
-    use configuration::NetworkConfigBuilder;
+    use configuration::{GlobalSettingsBuilder, NetworkConfigBuilder};
+    use lazy_static::lazy_static;
+    use tokio::sync::Mutex;
 
     use super::*;
 
+    const ENV_KEY: &str = "ZOMBIE_SPAWN_CONCURRENCY";
+    // mutex for test that use env
+    lazy_static! {
+        static ref ENV_MUTEX: Mutex<()> = Mutex::new(());
```

Comment:
Is this mutex really needed?
Isn't it enough to make sure that env is set/removed before `calculate_frequency` is called?

Reply:
Yes, since the unit test doesn't run in order (and can be run in parallel) we need to ensure the proper env is set.